### PR TITLE
#MAN-208 Code event/exhibit index page

### PR DESCRIPTION
**Pull request by cdoyle-temple**
#MAN-208

This design was reviewed with communications staff and agreed upon

***************************************

Add Kaminari for pagination (chose this gem for compatibility with Administrate)
Add event_type field to model since tags is not where Event Type comes from in the xml feed
Update event sync service to pull type field and map to event_type
Add promoted to events checkbox to exhibition model to pull only those approved to the events page
Serialized tags and types with methods in model
Added methods to model to call image variants from view
Updated routes to remove beyondthepage reference and revert

### CODE REVIEW

**Comment by cdoyle-temple** [comment]
This should be good now, as long as using Select is ok. 
We will have data issues due to the fact that the feed uses Paley Library and we don't have Paley in the new site. Spaces with buildings not in the site will appear as external spaces.

File: db/schema.rb [hunk 1]
```diff
@@ -171,8 +173,8 @@
     t.integer "person_id"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.index ["finding_aid_id"], name: "index_finding_aid_responsibilities_on_finding_aid_id"
-    t.index ["person_id"], name: "index_finding_aid_responsibilities_on_person_id"
+    t.index ["finding_aid_id"], name: "index_finding_aid_responsibility_on_finding_aid_id"
```
Comment (bibliotechy) [inline db/schema.rb]: Why did this change?

Comment (cdoyle-temple) [reply db/schema.rb]: It does not look there has been a change. The lines removed are added back just as they were, and in the same place. I.e. I don't know, but it shouldn't have any effect.

File: app/models/event.rb [hunk 2]
```diff
@@ -27,4 +37,15 @@ def set_times
       "All Day"
     end
   end
+
+  def index_image
+    variation =
+      ActiveStorage::Variation.new(Uploads.resize_to_fill(width: 220, height: 220, blob: image.blob, gravity: "Center"))
```
Comment (bibliotechy) [inline app/models/event.rb]: @cdoyle-temple I need you to walk me through what is happening here. If I understand this correctly, you are generating the image to that specified size each time. Whch means if there are 10 images on a page, we are generating them 10 times.

Comment (cdoyle-temple) [reply app/models/event.rb]: yes. is there a better way?

**Pull request merged**